src/ngraph_mark_for_clustering.cc

```diff
@@ -266,6 +266,10 @@ Status MarkForClustering(Graph* graph,
         TF_RETURN_IF_ERROR(GetNodeAttr(n->attrs(), "is_training", result));
         return Status::OK();
       };
+      confirmation_function_map["GatherV2"] = [](Node* n, bool* result) {
```

Comment:
If we move the getting of current backend before the mark for clustering you can use that in the confirmation function. Will that help ?